src/GraphQL/GraphQLValuesCache.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+using GraphQLParser;
+using GraphQLParser.AST;
+
+namespace GraphQL
+{
+    internal class GraphQLValuesCache
+    {
+        private static readonly Dictionary<ROM, object> _ints = new Dictionary<ROM, object>
+        {
+            ["-10"] = -10,
+            ["-9"] = -9,
+            ["-8"] = -8,
+            ["-7"] = -7,
+            ["-6"] = -6,
+            ["-5"] = -5,
+            ["-4"] = -4,
+            ["-3"] = -3,
+            ["-2"] = -2,
+            ["-1"] = -1,
+            ["0"] = 0,
+            ["1"] = 1,
+            ["2"] = 2,
+            ["3"] = 3,
+            ["4"] = 4,
+            ["5"] = 5,
+            ["6"] = 6,
+            ["7"] = 7,
+            ["8"] = 8,
+            ["9"] = 9,
+            ["10"] = 10,
+        };
+
+        private static readonly Dictionary<ROM, object> _longs = new Dictionary<ROM, object>
+        {
+            ["-10"] = -10L,
+            ["-9"] = -9L,
+            ["-8"] = -8L,
+            ["-7"] = -7L,
+            ["-6"] = -6L,
+            ["-5"] = -5L,
+            ["-4"] = -4L,
+            ["-3"] = -3L,
+            ["-2"] = -2L,
+            ["-1"] = -1L,
+            ["0"] = 0L,
+            ["1"] = 1L,
+            ["2"] = 2L,
+            ["3"] = 3L,
+            ["4"] = 4L,
+            ["5"] = 5L,
+            ["6"] = 6L,
+            ["7"] = 7L,
+            ["8"] = 8L,
+            ["9"] = 9L,
+            ["10"] = 10L,
+        };
+
+        public static object GetInt(ROM value) => _ints.TryGetValue(value, out object i) ? i : Int.Parse(value);
+
+        public static object GetLong(ROM value) => _longs.TryGetValue(value, out object i) ? i : Long.Parse(value);
```

Comment:
Are you sure using a dictionary here is best for performance vs an allocation?  I think dictionaries are a performance killer for this situation.  For example, a hash code now has to be generated for any value.  Usually I set a set of if conditions for both best performance.  I would suggest this logic:

1. if the length of the ROM is != 1 then parse value
2. if ROM == "0" return boxed 0
3. etc through "5"
4. else parse value

in this situation, if we know the length is 1 then we know it has to be 0-9 and we might as well continue with the if conditions through 9.  If we want to check for -1 then it's a check prior to the "if length is 1" check.

How does that sound?

Reply:
done